docs/DeveloperGuide.md

```diff
@@ -260,6 +266,87 @@ Note that only the `Category` field is allowed to be specified multiple times, w
+
+The activity diagram of deleting a Transaction is as shown below
+
+<img src="images/unicash/DeleteTransactionActivityDiagram.png" width="1200" />
+
+The following sequence diagram shows the interaction between different components of UniCash.
+
+<img src="images/unicash/DeleteTransactionSequenceDiagram.png" width="1200" />
+
+The above sequence diagram omits details on the filtering of `TransactionList` and assumes that
+the displayed `TransactionList` is showing all transactions. However, the logic of the `DeleteCommand`
+remains the same for all list deletion.
+
+ℹ️ **Note:** The lifeline for `DeleteTransactionCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML,
+the lifeline reaches the end of diagram.
+
+##### Details
+
+1. The user specifies the transaction to be deleted by stating the integer index of the transaction to be deleted.
+2. The input will be parsed by `DeleteTransactionCommandParser` and if the provided input is invalid, `ParseException` will be thrown,
+and the user is prompted to enter the command again with the correct input.
+3. If the input is valid, an `Index` object is created with the given input integer, and passed into `DeleteTransactionCommand` to be executed
+by `LogicManager`
+4. `LogicManger` will invoke the `execute` method of `DeleteTransactionCommand` which will delete the `Transaction` from UniCash.
+
+It is important to take note that when the user input is parsed, it is based on the currently displayed `TransactionList` inside
+`TransactionListPanel`. This means that even if a `TransactionList` contains `10` transactions, given a specific nominally valid
+number like `7`, it can still throw a `ParseException` if the shown `TransactionList` contains less than `7` items. This feature is
+intentional, as the User is able to, for example, `find` a particular group of transaction and immediately delete those transactions
+by just looking at their displayed index number without having to refer to an external identifier of that transaction. UniCash will
+automatically handle the visual ordering and representation of transactions with the `TransactionsListPanel` in the UI. The details
+and diagrams for this part will be elaborated further in the UI section (and other relevant sections) of this Developer Guide.
+
+
+### Feature Group 2 - Budget Management and Monitoring
+
+### Feature Group 3 - General Utility Features
+
+This includes commands such as Clear, Reset, Help and Exit.
+
+#### Clear Transactions
+
+##### Overview
+
+The `ClearTransactionsCommand` deletes all existing `Transactions` from `TransactionList` in UniCash.
+
+The activity diagram of clearing all transactions is as shown below
+
+<img src="images/unicash/ClearTransactionsActivityDiagram.png" width="1200" />
+
+The following sequence diagram shows the interaction between different components of UniCash.
+
+<img src="images/unicash/ClearTransactionsSequenceDiagram.png" width="1200" />
+
```

Comment:
I remember mentioned in tutorial the text in the images should be roughly the same as the text outside the image. Just a minor cosmetic issue. I think my PR also has this issue.

Reply:
Ohh kk sure thanks, I fixed it, I think it looks better now.